src/main/java/nl/ordina/jobcrawler/controller/SkillController.java

```diff
@@ -0,0 +1,123 @@
+        this.skillRepository = skillRepository;
+    }
+
+    // getting all skills from database
+    @GetMapping(path = "skills")
+    public List<Skill> getAllSkills() {
+        return skillService.findAll();
+    }
+
+    @GetMapping(path="getskills")
+    public List<SkillDTO> getAllMySkills() {
+        List<Skill> skills = skillRepository.findByOrderByNameAsc();
+
+        List<SkillDTO> skilldoaList = new ArrayList<>();
+        for (Skill s: skills) {
+            skilldoaList.add(new SkillDTO(s.getId().toString(), s.getName()));
+        }
+        return skilldoaList;
+    }
+
+    @PostMapping(path="saveskill")
+    /* Save a new skill created from the maintenance form in the database.
+
+       This funtion returns HttpStatus OK regardless of the outcome
+       Reason is that for example with HttpStatus UNPROCESSABLE_ENTITY, the front end Angular does
+       not "see" the error message
+
+       Response code values:
+       when no error occurs, error message will be "OK
+       when constraint violation occurs "Skill already exists" (a duplicate key is assumed)
+       in all other cases, the exception message from the repo
+    */
+    public ResponseEntity<ResponseCode> saveskill(@RequestBody SkillDTO skillDTO) {
+        log.info("save new skill:" + skillDTO.getName());
+
+        try {
+            if (skillDTO.getName().length() < 3) {
+                return new ResponseEntity<>(new ResponseCode("Name should be longer than 2 characters"),
+                        HttpStatus.OK);
+            }
+            skillService.save(new Skill(skillDTO.getName()));
+            return new ResponseEntity<>(new ResponseCode("OK"), HttpStatus.OK);
+        } catch (Exception e) {
+            ResponseEntity<ResponseCode> errorResult;
+            if (e instanceof DataIntegrityViolationException) {
+                 errorResult =
+                        new ResponseEntity<>(new ResponseCode("Skill already exists"), HttpStatus.OK);
+            } else {
+                String msg = e.getCause().getMessage();
+                errorResult =
+                        new ResponseEntity<>(new ResponseCode(msg), HttpStatus.OK);
+            }
+            return errorResult;
+        }
+    }
+
+
+    @PostMapping(path="deleteskill")
+    public ResponseEntity<ResponseCode> deleteskill(@RequestBody SkillDTO skillDTO) {
+        log.info("delete skill:" + skillDTO.getName());
+        try {
+            skillRepository.deleteSkillByName(skillDTO.getName());
+//            skillService.deleteSkill(skillDTO.getName());
+            return new ResponseEntity<>(new ResponseCode("OK"), HttpStatus.OK);
+        } catch (Exception e) {
+            return new ResponseEntity<>(
+                    new ResponseCode("Could not delete skill:" + e.getLocalizedMessage()), HttpStatus.OK);
+        }
+    }
+
+    // getting all skills from database
+    @GetMapping(path = "relinkskills")
+    public ResponseEntity<ResponseCode> relinkSkills() {
```

Comment:
this comment does not describe the method below it

Reply:
will adapt